A more efficient mongodb query; $or was executing multiple queries in

sonicblend opened the pull request:
parallel. I've also added delay_until to the hash that gets inserted,
to further tidy the search query.

Commit:
A more efficient mongodb query; $or was executing multiple queries in
parallel. I've also added delay_until to the hash that gets inserted,
to further tidy the search query.

ian-kent commented on lib/MangoX/Queue.pm:
```diff
@@ -85,24 +85,21 @@ sub get_options {
-            }]
+            status => {
+                '$in' => [
+                    ref($self->{pending_status}) eq 'ARRAY' ? @{$self->{pending_status}} : $self->{pending_status},
```
one tiny comment :)

```
[
    ref($self->{pending_status}) eq 'ARRAY' ? @{$self->{pending_status}} : $self->{pending_status}, $self->{processing_status},
]
```

means a new array ref will be constructed every time get_options is called, and `$self->{pending_status}` will be dereferenced to put the values back in the other array ref

since we're now always bringing in an additional status (pending_status and processing_status), it'd be better if we create the array ref up-front and reuse it - maybe by combining pending_status and processing_status parameters when the object is instantiated (we might then also need new methods to set pending_status/processing_status at runtime to build the new array ref)

sonicblend replied:
Good idea, see my latest commit.

Commit:
Add failed_status, pending_status & processing_status attributes rather
than directly modifying the object hash. Enables the status array to be
pre-generated rather than on the fly when get_options is called.

Commit:
mention that delay_until queue key is now required
